packages/headless/src/composables/slider/createSlider.ts

```diff
@@ -0,0 +1,861 @@
+        const next = values[nextIdx];
+        if (next !== value && typeof next === "number") commitValueFromEvent(event, next);
+        return;
+      }
+    }
+  };
+
+  const trackOffset = computed(() =>
+    MathUtils.valueToPercent(
+      isRange.value && values.value[0] ? values.value[0] : min.value,
+      min.value,
+      max.value,
+    ),
+  );
+  const trackLength = computed(
+    () =>
+      MathUtils.valueToPercent(values.value.at(-1) ?? 0, min.value, max.value) - trackOffset.value,
+  );
+  const trackStyle = computed(() => ({
+    [axis.value.position]: `${trackOffset.value}%`,
+    [axis.value.size]: `${trackLength.value}%`,
+  }));
+
+  const isMarkActive = (markValue: number) => {
+    if (isRange.value) {
+      const minValue = Math.min(...values.value);
+      const maxValue = Math.max(...values.value);
+      return markValue >= minValue && markValue <= maxValue;
+    }
+
+    const currentValue = values.value[0]!;
+    return markValue <= currentValue;
+  };
+
+  onBeforeUnmount(stopPointerListening);
+
+  watch(
+    () => isDisabled.value,
+    () => {
+      if (isDisabled.value) {
+        isDragging.value = false;
+        activeThumbIndex.value = -1;
+        focusedThumbIndex.value = -1;
+        stopPointerListening();
+      }
+    },
+  );
+
+  return {
+    elements: {
+      /**
+       * Root slider container element
+       */
+      root: computed(() => ({
+        ref: sliderRef,
+        style: { touchAction: orientation.value === "vertical" ? "pan-x" : "pan-y" },
+        onMousedown: handleRootMousedown,
+        onTouchstart: handlePointerStart,
+      })),
+
+      /**
+       * Individual thumb elements for each value (span)
+       */
+      thumbContainer: computed(() => (data: { index: number; value: number }) => ({
+        "data-index": data.index,
+        style: {
+          [axis.value.position]: `${MathUtils.valueToPercent(data.value, min.value, max.value)}%`,
+        },
+      })),
+
+      /**
+       * Visually hidden input inside each thumb for accessibility
+       */
+      thumbInput: computed(() => (data: { index: number; value: number }) => ({
+        min: min.value,
+        max: max.value,
+        value: data.value,
+        role: "slider",
+        type: "range",
+        "aria-label": label.value,
+        "aria-valuemin": min.value,
+        "aria-valuemax": max.value,
+        "aria-valuenow": data.value,
+        "aria-orientation": orientation.value,
+        "data-index": data.index,
+        tabIndex: isDisabled.value ? -1 : 0,
+        step: step.value === null && marks.value ? "any" : (step.value ?? undefined),
+        disabled: typeof isDisabled.value === "boolean" ? isDisabled.value : false,
+        onChange: handleHiddenInputChange,
+        onFocus: handleHiddenInputFocus,
+        onBlur: handleHiddenInputBlur,
+        onKeydown: handleHiddenInputKeydown,
+        /**
+         * Visually hidden styles.
+         */
+        style: {
```

Comment:
The styling should be responsibility of the component, not the headless logic :) onyx has a OnyxVisuallyHidden component that you can use for this